src/main/webapp/app/exercises/programming/manage/update/update-components/programming-exercise-language.component.html

```diff
@@ -0,0 +1,84 @@
+<div class="form-group">
+    <label class="label-narrow" jhiTranslate="artemisApp.programmingExercise.programmingLanguage" for="field_programmingLanguage">Programming Language</label>
+    <select
+        #select
+        required
+        class="form-select"
+        [ngModel]="languageStepInputs.selectedProgrammingLanguage"
+        (ngModelChange)="select.value = languageStepInputs.onProgrammingLanguageChange($event)"
+        name="programmingLanguage"
+        id="field_programmingLanguage"
+        [disabled]="isImport || !!programmingExercise.id"
+    >
+        <option value="JAVA" *ngIf="languageStepInputs.supportsJava">Java</option>
```

Comment:
Is it possibly to provide an array of supported languages and iterate over them?

```ts
// In a sensible location
const supportedLanguages = ["java", "python", etc]
```

```html
<option *ngFor="let supportedLanguage of supportedLanguages" value="DERIVE THAT SENSIBLY FROM VALUE">{{ supportedLanguage }}</option>
```